File: 12. Rekursi/D. Kasur Rusak/d_kasur_rusak.pas
program d_kasur_rusak;

function palindrome(str: string): boolean;
var
    len: integer;
    sub: boolean;
begin
    len := length(str);
    if len = 1 then
        palindrome := true
    else if len = 2 then
        palindrome := str[1] = str[2]
    else begin
        if str[1] = str[len] then
            sub := palindrome(copy(str, 2, len - 2))
        else
            sub := false;
        palindrome := sub;
    end;
end;

var
    S: string;
begin
    readln(S);
    if palindrome(S) then
        writeln('YA')
    else
        writeln('BUKAN');
end.

File: 12. Rekursi/D. Kasur Rusak/d_kasur_rusak.py
def palindrome(S):
    if len(S) == 1:
        return True
    elif len(S) == 2:
        return S[0] == S[1]
    else:
        return S[0] == S[-1] and palindrome(S[1:-1])

S = input()
print("YA") if palindrome(S) else print("BUKAN")
